perfkitbenchmarker/providers/gcp/gce_virtual_machine.py

```diff
@@ -106,6 +106,72 @@ def __init__(self, vm_spec):
 
     events.sample_created.connect(self.AnnotateSample, weak=False)
 
+  def _GenerateCreateCommand(self, ssh_keys_path):
+    """Generates a command to create the VM instance.
+
+    Args:
+      ssh_keys_path: string. Path to a file containing the sshKeys metadata.
+
+    Returns:
+      GcloudCommand. gcloud command to issue in order to create the VM instance.
+    """
+    cmd = util.GcloudCommand(self, 'compute', 'instances', 'create', self.name)
+    cmd.flags['image'] = self.image
+    cmd.flags['boot-disk-size'] = self.BOOT_DISK_SIZE_GB
+    cmd.flags['boot-disk-type'] = self.BOOT_DISK_TYPE
+    cmd.flags['machine-type'] = self.machine_type
+    cmd.flags['tags'] = 'perfkitbenchmarker'
+    cmd.flags['no-restart-on-failure'] = True
+    cmd.flags['metadata-from-file'] = 'sshKeys=%s' % ssh_keys_path
+    metadata = ['owner=%s' % FLAGS.owner]
+    for key, value in self.boot_metadata.iteritems():
+      metadata.append('%s=%s' % (key, value))
+    cmd.flags['metadata'] = ','.join(metadata)
+    if not FLAGS.gce_migrate_on_maintenance:
+      cmd.flags['maintenance-policy'] = 'TERMINATE'
+    ssd_interface_option = NVME if NVME in self.image else SCSI
+    cmd.flags['local-ssd'] = (['interface={0}'.format(ssd_interface_option)] *
+                              self.max_local_disks)
+    if FLAGS.gcloud_scopes:
+      cmd.flags['scopes'] = ','.join(re.split(r'[,; ]', FLAGS.gcloud_scopes))
+    if self.preemptible:
+      cmd.flags['preemptible'] = True
+    return cmd
+
+  def _GenerateDescribeCommand(self):
+    """Generates a command to describe the VM instance.
+
+    Returns:
+      GcloudCommand. gcloud command to issue in order to describe the VM
+      instance.
+    """
+    return util.GcloudCommand(self, 'compute', 'instances', 'describe',
+                              self.name)
+
+  def _GenerateDeleteCommand(self):
+    """Generates a command to delete the VM instance.
+
+    Returns:
+      GcloudCommand. gcloud command to issue in order to delete the VM instance.
+    """
+    return util.GcloudCommand(self, 'compute', 'instances', 'delete', self.name)
```

Comment:
Nit: I think all but `Create` of these `_Generate*Command` functions could be inlined where used - they're ~7 lines of declaration / docstring for a few lines of actual code.